featureCounts-combine/main.nf

```diff
@@ -0,0 +1,29 @@
+process RUN_FEATURECOUNTS_COMBINE {
+    
+    cpus 1
+    memory "16.G"
+    time "10.m"
+    publishDir "results/${params.project}", mode: 'copy'
+
+    input:
+    path(counts)
+
+    output:
+    path(count_table)
+
+    script:
+    count_table = "${params.project}_counts.txt"
+    """
+    # Get the gene IDs 
+    cut -f1 ${counts[0]} > ensg.txt
+
+    # store the counts without IDs in a tmp file for each
+    for file in $counts; do
+        cut -f2 \$file > \${file}.tmp
+    done
+
+    # put them all into a single file
+    paste ensg.txt *txt.tmp > $count_table
+    """
+
```

Comment:
@jarrydmartinx is there anything wildly stupid looking about this bash code? it is just supposed to take a set of input files as a list and combine them into a single file with n+1 columns where n is the number of samples. If there is a much better way please suggest! This is just the first option I came up with (we can safely assume that the geneid columns are the same order for all samples)

input file format

Geneid	sample_1
A	1
B	2
C	3

output file format

Geneid	sample_1	sample_2	sample_n
A	1	0	4	1
B	2	3	2	1
C	3	0	1	1